packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts

```diff
@@ -0,0 +1,460 @@
+import { beforeEach, describe, it, expect } from "vitest";
+import { createMyTestRunner, createTcgcContext } from "./test-hots.js";
+import { serializeArray } from "../../../src/modular/serialization/serializeArray.js";
+import {
+  SdkArrayType,
+  SdkContext
+} from "@azure-tools/typespec-client-generator-core";
+import { EmitContext, UsageFlags } from "@typespec/compiler";
+import { BasicTestRunner } from "@typespec/compiler/testing";
+import { SerializeTypeOptions } from "../../../src/modular/serialization/serializers.js";
+import { format } from "prettier";
+import { SerializerMap } from "../../../src/modular/serialization/util.js";
+
+describe("serializeArray", () => {
+  let runner: BasicTestRunner;
+
+  beforeEach(async () => {
+    runner = await createMyTestRunner();
+  });
+
+  describe("serialization", () => {
+    let testArrays: TestArrays;
+    beforeEach(async () => {
+      testArrays = await getSdkArrays(runner, "serialize");
+    });
+
+    it("should handle a string array", async () => {
+      const { stringArray } = testArrays;
+      const result = serializeArray(stringArray);
+
+      expect(result).to.equal("myArray");
+    });
+
+    it("should handle a numeric array", async () => {
+      const { numericArray } = testArrays;
+      const result = serializeArray(numericArray);
+
+      expect(result).to.equal("myArray");
+    });
+
+    it("should handle a boolean array", async () => {
+      const { booleanArray } = testArrays;
+      const result = serializeArray(booleanArray);
+
+      expect(result).to.equal("myArray");
+    });
+
+    it("should handle a date array", async () => {
+      const { dateArray } = testArrays;
+      const result = serializeArray(dateArray);
+
+      expect(result).to.equal("myArray.map((e: Date)=>((e).toISOString()))");
+    });
+
+    it("should handle a model array inline", async () => {
+      const { modelArray } = testArrays;
+      const result = serializeArray({
+        ...modelArray,
+        serializerMap: undefined
+      });
+
+      const expected = await format(
+        `myArray.map((e: Foo)=>({...(e),
+                            "bar": ({...(e["bar"]),
+                            "baz": ({...(e["bar"]["myBaz"]),
+                            "qux": (e["bar"]["myBaz"]["qux"])})})}))`,
+        { parser: "typescript" }
+      );
+
+      const actual = await format(result, { parser: "typescript" });
+
+      expect(actual).to.equal(expected);
+    });
+
+    it("should handle a model array ", async () => {
+      const { modelArray } = testArrays;
+      const result = serializeArray(modelArray);
+
+      const expected = await format(`myArray.map((e) => serializeFoo(e));;`, {
```

Comment:
Will the formatter catch this duplicated semicolon? Or is that expected?

Reply:
Yeah it is a typo on my side, formatter will remove it. I'll fix it in the next commit